sg/logger.go

```diff
@@ -24,8 +24,10 @@ func WithLogger(ctx context.Context, logger *log.Logger) context.Context {
 
 // Logger returns the log.Logger attached to ctx, or a default logger.
 func Logger(ctx context.Context) *log.Logger {
-	if logger := ctx.Value(loggerContextKey{}).(*log.Logger); logger != nil {
-		return logger
+	if ok := ctx.Value(loggerContextKey{}); ok != nil {
+		if logger := ok.(*log.Logger); logger != nil {
```

Comment:
This should not be necessary since it's our own private functions that assign the logger value - it's impossible to get any other type than `*log.Logger` out.

Reply:
That’s what I thought, however i get a panic without this 🤷‍♀️

Reply:
But I guess I could try and find out why

Reply:
hmm yes, could you post a stack trace?